packages/vite/src/node/optimizer/esbuildDepPlugin.ts

```diff
@@ -145,7 +145,7 @@ export function esbuildDepPlugin(
         const entryFile = qualified[id]
 
         let relativePath = normalizePath(path.relative(root, entryFile))
-        if (!relativePath.startsWith('.')) {
+        if (!relativePath.startsWith('./')) {
```

Comment:
Just to check, could the `relativePath` be just a dot in this context? Maybe to get the index.js in the root folder? If that is the case we should also check for it.
```suggestion
        if (!relativePath.startsWith('./') && relativePath !== '.') {
```